apps/etterlatte-pdltjenester/src/main/kotlin/pdl/PdlOboKlient.kt

```diff
@@ -0,0 +1,88 @@
+
+import com.github.michaelbull.result.get
+import com.typesafe.config.Config
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.accept
+import io.ktor.client.request.bearerAuth
+import io.ktor.client.request.header
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import no.nav.etterlatte.libs.common.RetryResult
+import no.nav.etterlatte.libs.common.innsendtsoeknad.common.Behandlingsnummer
+import no.nav.etterlatte.libs.common.retry
+import no.nav.etterlatte.libs.ktorobo.AzureAdClient
+import no.nav.etterlatte.token.BrukerTokenInfo
+import org.slf4j.LoggerFactory
+
+class PdlOboKlient(private val httpClient: HttpClient, private val config: Config) {
+    private val logger = LoggerFactory.getLogger(PdlKlient::class.java)
+
+    private val apiUrl = config.getString("pdl.url")
+    private val pdlScope = config.getString("pdl.scope")
+
+    private val azureAdClient = AzureAdClient(config, httpClient)
+
+    suspend fun hentPersonNavn(
+        ident: String,
+        bruker: BrukerTokenInfo,
+    ): PdlPersonResponse {
+        val request =
+            PdlGraphqlRequest(
+                query = getQuery("/pdl/hentPerson.graphql"),
+                variables = PdlVariables(ident),
+            )
+
+        return retry<PdlPersonResponse>(times = 3) {
+            httpClient.post(apiUrl) {
+                bearerAuth(getOboToken(bruker))
+                header(HEADER_BEHANDLINGSNUMMER, Behandlingsnummer.BARNEPENSJON.behandlingsnummer)
+                header(HEADER_TEMA, HEADER_TEMA_VALUE)
+                contentType(ContentType.Application.Json)
+                accept(ContentType.Application.Json)
+                setBody(request)
+            }.body()
+        }.let {
+            when (it) {
+                is RetryResult.Success ->
+                    it.content.also { result ->
+                        result.errors?.joinToString()?.let { feil ->
+                            logger.error("Fikk data fra PDL, men også følgende feil: $feil")
+                        }
+                    }
+
+                is RetryResult.Failure -> throw it.samlaExceptions()
+            }
+        }
+    }
+
+    private fun getQuery(name: String): String {
+        return javaClass.getResource(name)!!
+            .readText()
+            .replace(Regex("[\n\t]"), "")
+    }
+
+    private suspend fun getOboToken(bruker: BrukerTokenInfo): String {
+        val token =
+            azureAdClient.getOnBehalfOfAccessTokenForResource(
+                listOf(pdlScope),
+                bruker.accessToken(),
+            )
+
+        return requireNotNull(token.get()?.accessToken) {
+            "Kunne ikke hente ut obo-token for bruker ${bruker.ident()}"
+        }
+    }
+
+    /**
+     * TODO: Vi må ta stilling til hvilke behandlingsnummer og tema vi skal bruke på "generelle" forespørsler.
```

Comment:
Hva bruker du nå?

Reply:
Mao, endre i behandlingskatalogen også kan vi merge inn denne

Reply:
Vi burde nok hatt ett nytt behandlingsnummer for akkurat det her

Reply:
Fikset. Ref. tråd på slack bruker vi bare begge behandlingsnummerene våre.